src/main/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistency.java

```diff
@@ -12,10 +15,11 @@
 
     private AuroraMultiReplicaConsistency(
         ReplicaConsistency replicaConsistency,
-        AuroraConnectionDetails auroraConnectionDetails
+        AuroraConnectionDetails auroraConnectionDetails,
+        BiFunction<String, Properties, AuroraConnectionSupplier> connectionPerReplicaProvider
```

Comment:
Can we avoid BiFunction in the API? It's hard to use, and it blocks future API evolution.

Reply:
extracted the function into a separate interface